### PR TITLE
TestingNode

### PR DESCRIPTION
**Pull request by camilamaia**
## Description

TestingNode:

```yaml
api:
  endpoints:
    - name: scanapi-demo
      path: ${BASE_URL}
      requests:
        - name: health
          method: get
          path: /health/
          tests:
            - name: Response Status is 200
              assert: ${{ response.status_code == 200 }}
            - name: Response Body Equals OK
              assert: ${{ response.content == b"OK!" }}
        - name: languages
          method: get
          path: /languages/
          tests:
            - name: Response Status is 200
              assert: ${{ response.status_code == 200 }}
      endpoints:
        - name: devs
          path: /devs/
          requests:
            - name: list_all
              method: get
              vars:
                uuid: ${{response.json()[2]["uuid"]}}
              tests:
                - name: Response Status is 200
                  assert: ${{ response.status_code == 200 }}
            - name: looking_for_new_opportunities
              method: get
              params:
                newOpportunities: true
              tests:
                - name: Response Status is 200
                  assert: ${{ response.status_code == 200 }}
            - name: not_looking_for_new_opportunities
              method: get
              params:
                newOpportunities: false
              tests:
                - name: Response Status is 200
                  assert: ${{ response.status_code == 200 }}
            - name: new
              method: post
              headers:
                x-api-key: ${DEMO_KEY}
              tests:
                - name: Response Status is 201
                  assert: ${{ response.status_code == 201 }}
              body:
                uuid: ${{uuid.uuid4().hex}}
                name: Tarik
                yearsOfExperience: 2
                languages:
                  - ruby
                    go
                newOpportunities: false
            - name: details_not_found
              path: 129e8cb2-d19c-51ad-9921-cea329bed7fa
              method: get
              tests:
                - name: Response Status is 404
                  assert: ${{ response.status_code == 404 }}
          endpoints:
            - name: details
              path: ${uuid}
              requests:
                - name: get
                  method: get
                  tests:
                    - name: Response Status is 200
                      assert: ${{ response.status_code == 200 }}
                - name: delete
                  method: delete
                  headers:
                    x-api-key: ${DEMO_KEY}
                  tests:
                    - name: Response Status is 200
                      assert: ${{ response.status_code == 200 }}
                - name: languages
                  path: languages
                  method: get
                  tests:
                    - name: Response Status is 200
                      assert: ${{ response.status_code == 200 }}
```

When tests passed:

```shell
Loading file .scanapi.yaml
Loading file api.yaml
Writing documentation
Making request GET http://demo.scanapi.dev/api/health/
 [PASSED] scanapi-demo::health::Response Status is 200
 [PASSED] scanapi-demo::health::Response Body Equals OK
Making request GET http://demo.scanapi.dev/api/languages/
 [PASSED] scanapi-demo::languages::Response Status is 200
Making request GET http://demo.scanapi.dev/api/devs/
 [PASSED] scanapi-demo::devs::list_all::Response Status is 200
Making request GET http://demo.scanapi.dev/api/devs/
 [PASSED] scanapi-demo::devs::looking_for_new_opportunities::Response Status is 200
Making request GET http://demo.scanapi.dev/api/devs/
 [PASSED] scanapi-demo::devs::not_looking_for_new_opportunities::Response Status is 200
Making request POST http://demo.scanapi.dev/api/devs/
 [PASSED] scanapi-demo::devs::new::Response Status is 201
Making request GET http://demo.scanapi.dev/api/devs/129e8cb2-d19c-51ad-9921-cea329bed7fa
 [PASSED] scanapi-demo::devs::details_not_found::Response Status is 404
Making request GET http://demo.scanapi.dev/api/devs/129e8cb2-d19c-41ad-9921-cea329bed7f0
 [PASSED] scanapi-demo::devs::details::get::Response Status is 200
Making request DELETE http://demo.scanapi.dev/api/devs/129e8cb2-d19c-41ad-9921-cea329bed7f0
 [PASSED] scanapi-demo::devs::details::delete::Response Status is 200
Making request GET http://demo.scanapi.dev/api/devs/129e8cb2-d19c-41ad-9921-cea329bed7f0/languages
 [PASSED] scanapi-demo::devs::details::languages::Response Status is 200

The documentation was generated successfully.
It is available at scanapi-report.md
```

When a test fails:

```shell
Loading file .scanapi.yaml
Loading file api.yaml
Writing documentation
Making request GET http://demo.scanapi.dev/api/health/
 [PASSED] scanapi-demo::health::Response Status is 200
 [PASSED] scanapi-demo::health::Response Body Equals OK
Making request GET http://demo.scanapi.dev/api/languages/
 [PASSED] scanapi-demo::languages::Response Status is 200
Making request GET http://demo.scanapi.dev/api/devs/
 [PASSED] scanapi-demo::devs::list_all::Response Status is 200
Making request GET http://demo.scanapi.dev/api/devs/
 [PASSED] scanapi-demo::devs::looking_for_new_opportunities::Response Status is 200
Making request GET http://demo.scanapi.dev/api/devs/
 [PASSED] scanapi-demo::devs::not_looking_for_new_opportunities::Response Status is 200
Making request POST http://demo.scanapi.dev/api/devs/
 [PASSED] scanapi-demo::devs::new::Response Status is 201
Making request GET http://demo.scanapi.dev/api/devs/129e8cb2-d19c-51ad-9921-cea329bed7fa
 [PASSED] scanapi-demo::devs::details_not_found::Response Status is 404
Making request GET http://demo.scanapi.dev/api/devs/129e8cb2-d19c-41ad-9921-cea329bed7f0
 [FAILED] scanapi-demo::devs::details::get::Response Status is 200
	  response.status_code == 300 is false
Making request DELETE http://demo.scanapi.dev/api/devs/129e8cb2-d19c-41ad-9921-cea329bed7f0
 [PASSED] scanapi-demo::devs::details::delete::Response Status is 200
Making request GET http://demo.scanapi.dev/api/devs/129e8cb2-d19c-41ad-9921-cea329bed7f0/languages
 [PASSED] scanapi-demo::devs::details::languages::Response Status is 200

The documentation was generated successfully.
It is available at scanapi-report.md
```

Closes https://github.com/scanapi/scanapi/issues/11

Next Steps:

- Save tests information in a session.
- Exit the program with an error when a test fails
- Move information about tests to console/markdown/html reports

### CODE REVIEW

File: scanapi/tree/testing_node.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    def assertion(self):
+        return self["assert"]
+
+    def run(self):
+        passed, error = self.request.endpoint.vars.evaluate(
+            self.assertion, is_a_test_case=True
+        )
+        passed = "PASSED" if passed else "FAILED"
+        print(
+            f"[{passed}] {self.request.endpoint.name}::{self.request.name}::{self.name}"
```
Comment (camilamaia) [inline scanapi/tree/testing_node.py:39]: We need to iterate here to check if there are more parent endpoints

File: pytest.ini [hunk 2]
```diff
@@ -1,3 +1,2 @@
 [pytest]
 addopts=-svvl
-bdd_features_base_dir = tests/functional/features/
```
Comment (camilamaia) [inline pytest.ini:3]: We are not using pytest-bdd anymore

**Comment by camilamaia** [comment]
Codecov report: https://codecov.io/github/scanapi/scanapi/commit/11088d4124e876aa350e8d105c53964e5c488c1d

File: scanapi/evaluators/code_evaluator.py [hunk 3]
```diff
@@ -12,29 +19,45 @@ class CodeEvaluator:
...
+        :type code: string
+        :param response: the response for the current request that is being tested
+        :type response: requests.Response
+        :return: A boolean that indicates if assert is True/False and, if False, the code tested.
+        :rtype: (boolean, string)
+        """
+        try:
+            assert eval(code)
+            return (True, None)
+        except AssertionError as ae:
```
Comment (winstonf88) [inline scanapi/evaluators/code_evaluator.py:53]:
```suggestion
        except AssertionError:
```

File: scanapi/tree/endpoint_node.py [hunk 4]
```diff
@@ -51,7 +51,12 @@ def __repr__(self):
 
     @property
     def name(self):
-        return self.spec.get(NAME_KEY, ROOT_SCOPE)
+        name = self.spec.get(NAME_KEY, "")
+
+        if self.is_root or not self.parent.name:
+            return name
+
+        return "::".join((self.parent.name, name))
```
Comment (winstonf88) [inline scanapi/tree/endpoint_node.py:59]:
```suggestion
        return f"{self.parent.name}::{name}"
```

File: scanapi/evaluators/string_evaluator.py [hunk 5]
```diff
@@ -15,7 +15,7 @@ class StringEvaluator:
     )  # ${<variable>}
 
     @classmethod
-    def evaluate(cls, sequence, vars):
+    def evaluate(cls, sequence, vars, is_a_test_case=False):
```
Comment (winstonf88) [inline scanapi/evaluators/string_evaluator.py:18]: this `is_a_test_case` looks kind of scattered all over

Comment (camilamaia) [reply scanapi/evaluators/string_evaluator.py:18]: @winstonf88 Hmmm indeed... Any ideas of how to minimize it?

Comment (camilamaia) [reply scanapi/evaluators/string_evaluator.py:18]: I tried to isolate it a bit more in a separated `evaluate_assertion` method, but we can do it better in the future for sure.

Comment (winstonf88) [reply scanapi/evaluators/string_evaluator.py:18]: Really cool! it's much more isolated. Great job.

Comment (camilamaia) [reply scanapi/evaluators/string_evaluator.py:18]: Thanks, man 💚

**Pull request merged**